src/parser.js

```diff
@@ -194,12 +194,17 @@ function runInSandbox({ markup, query, cacheId }) {
         rootNode: sandbox.rootNode,
         element,
       });
-
+      const cssPathString = cssPath(result.data, true);
+      const container = document.createElement('div');
+      container.innerHTML = markup;
+      const target = container.querySelector(
+        cssPathString.replace(/^body > /, ''),
+      );
```

Comment:
This should be done in `ResultQueries.js` instead. Close to the using code.

Reply:
I have added there because It could be useful in other parts

Reply:
Can we not just call the `getSuggestedQuery` method here than? And only save the result? 

Then we could also stop returning `target` from the parser.

Reply:
But we need to do that for every query. 
In the parser could we add a new object with all the methods ?

Reply:
Yes, that's why I meant. Return all suggestions directly from the parser. Something like:

```js
return {
  elements: [{
    queries: {
      role: getSuggestedQuery(target, 'get', 'Role'), 
      labelText:... 
  }], 
} 
```

Reply:
Yes it will work :)